### PR TITLE
Added GameModeSelection and CameraSubmersionEnvironment

### PR DESCRIPTION
**Pull request by mounderfod**
This is my first PR to yarn, please let me know if I have missed any conventions so that I can amend them. The PR adds mappings for two of the new snapshot enums: `CameraSubmersionEnvironment` and `GameModeSelection`.
Thanks :)

### CODE REVIEW

File: mappings/net/minecraft/client/render/ClientCauldronFluid.mapping [hunk 1]
```diff
@@ -0,0 +1,5 @@
+CLASS net/minecraft/class_5636 net/minecraft/client/render/ClientCauldronFluid
```
Comment (haykam821) [inline mappings/net/minecraft/client/render/ClientCauldronFluid.mapping]: This class isn't related to cauldrons in any way.

**Review by haykam821 — CHANGES_REQUESTED** [review]

File: mappings/net/minecraft/client/render/ClientCauldronFluid.mapping (hunk 1 above)
Comment (mounderfod) [reply mappings/net/minecraft/client/render/ClientCauldronFluid.mapping]: Solved in `f84bdc7`

**Review by YanisBft — APPROVED** [review]

**Review by Juuxel — CHANGES_REQUESTED** [review]
`Camera.getSubmergedFluidState` should be renamed to match the new enum, but this looks good otherwise.

**Comment by mounderfod** [comment]
> `Camera.getSubmergedFluidState` should be renamed to match the new enum, but this looks good otherwise.

@Juuxel Should I make this change myself, or should it be in another PR?

**Comment by Juuxel** [comment]
It should be in this PR since it's very related. (the other name makes no sense if it's not changed)

**Review by Juuxel — APPROVED** [review]

**Review by MarthDaul — APPROVED** [review]
No issues found.

**Review by Gegy — APPROVED** [review]

**Review by modmuss50 — APPROVED** [review]

**Pull request merged**